### PR TITLE
LOOP-2339: Nov DIY Merge

### PR DESCRIPTION
**Pull request by ps2**
https://github.com/tidepool-org/LoopKit/pull/277

* updates for skeleton onboarding (user should configure therapy settings before using Loop)
* Moving Tidepool adverse event reporting into Tidepool Service

### CODE REVIEW

**Review by nhamming — APPROVED** [review]
LGTM

**Comment by nhamming** [comment]
@ps2 I just noticed that the testing for the adverse event reporting was dropped. I assume it was on purpose (not sure why... but the testing was pretty simple at this stage), but wanted to make sure it wasn't overlooked.

**Comment by ps2** [comment]
@nhamming Thanks, not on purpose; meant to move it into TidepoolService. Will take a look.

**Comment by ps2** [comment]
@nhamming Added the tests back. Thanks for catching that.

**Review by nhamming — APPROVED** [review]
Looks good.

File: TidepoolService.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1294,6 +1307,7 @@
 		A9DAAD3022E7988900E76C9F /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ENABLE_MODULES = YES;
```
Comment (darinkrauss) [inline TidepoolService.xcodeproj/project.pbxproj:1310]: [nit] Please revert. These settings are defined at the project level. (I've been trying to keep all common settings at the project level and limit target specific settings to just those that are unique for the target.)

File: TidepoolService.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1304,13 +1318,16 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = org.tidepool.TidepoolServiceKitUITests;
 				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
```
Comment (darinkrauss) [inline TidepoolService.xcodeproj/project.pbxproj:1321]: [nit] Please revert. These settings are defined at the project level. (I've been trying to keep all common settings at the project level and limit target specific settings to just those that are unique for the target.)

File: TidepoolService.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1304,13 +1318,16 @@
...
+				SWIFT_VERSION = 5.0;
 				TARGETED_DEVICE_FAMILY = "1,2";
 			};
 			name = Debug;
 		};
 		A9DAAD3122E7988900E76C9F /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ENABLE_MODULES = YES;
```
Comment (darinkrauss) [inline TidepoolService.xcodeproj/project.pbxproj:1330]: [nit] Please revert. These settings are defined at the project level. (I've been trying to keep all common settings at the project level and limit target specific settings to just those that are unique for the target.)

File: TidepoolService.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1321,6 +1338,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = org.tidepool.TidepoolServiceKitUITests;
 				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_VERSION = 5.0;
```
Comment (darinkrauss) [inline TidepoolService.xcodeproj/project.pbxproj:1341]: [nit] Please revert. These settings are defined at the project level. (I've been trying to keep all common settings at the project level and limit target specific settings to just those that are unique for the target.)

**Review by darinkrauss — APPROVED** [review]
One comment (in 4 places) that I'd like changed, but otherwise LGTM! 🚀

**Review by rickpasetto — APPROVED** [review]
LGTM

**Pull request merged**